ipapython/ipaldap.py

```diff
@@ -1156,7 +1199,8 @@ def _connect(self):
             if not self._sasl_nocanon:
                 conn.set_option(ldap.OPT_X_SASL_NOCANON, ldap.OPT_OFF)
 
-            if self._start_tls:
+            if self._start_tls and self.protocol == 'ldap':
+                # STARTTLS applies only to ldap:// connections
```

Comment:
Should this be an assert in the constructor instead to prevent this sort of bogus combination?